src/components/catalogue/NumberInputComponent.svelte

```diff
@@ -8,28 +8,37 @@
 
     let {
         element,
+        inSearchBar = false,
+        resetToEmptySearchBar = () => {},
```

Comment:
To decouple this from the search bar I think it would be nicer to expose an event handler `onItemAdded` here and then register the callback from outside:

```html
<NumberInputComponent onItemAdded={resetToEmptySearchBar}>
```